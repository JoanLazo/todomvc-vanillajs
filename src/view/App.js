import Header from './Header.js';
import Todos from './Todos.js';
import Footer from './Footer.js';

/**
 * Crea el html de la app dinamicamente basandose
 * en los `todos` que se reciben como parametro
 *
 * @param todos Array de objetos con los todos existentes
 */
export default (todos) => {
  // Creamos cada una de las secciones
  const headerSrc = Header();
  const todosSrc = Todos(todos);
  const footerSrc = Footer(todos.length);

  // Creamos el elemento padre con template string
  const appSrc = `
    <section class="todoapp">
      ${headerSrc}
      ${todosSrc}
      ${footerSrc}
    </section>
  `;

  return appSrc;
};
