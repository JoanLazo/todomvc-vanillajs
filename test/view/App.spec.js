import App from '../../src/view/App';

describe(App, () => {
  it('Se pinta correctamente', () => {
    const todos = [
      { id: 1, label: 'todo 1' },
      { id: 2, label: 'todo 2' },
    ];
    const appElem = App(todos);
    // Mas info de snapshots en https://jestjs.io/docs/es-ES/snapshot-testing
    expect(appElem.outerHTML).toMatchSnapshot();
  });
});
