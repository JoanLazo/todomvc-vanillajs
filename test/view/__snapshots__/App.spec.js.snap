// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_default Se pinta correctamente 1`] = `
"<section class=\\"todoapp\\"><header class=\\"header\\">
    <h1>todos</h1>
    <input class=\\"new-todo\\" placeholder=\\"What needs to be done?\\" autofocus=\\"\\">
  </header><section class=\\"main\\"><div class=\\"todo-list\\"><li data-id=\\"1\\">
    <div class=\\"view\\">
      <label>todo 1</label>
      <button class=\\"destroy\\"></button>
    </div>
  </li><li data-id=\\"2\\">
    <div class=\\"view\\">
      <label>todo 2</label>
      <button class=\\"destroy\\"></button>
    </div>
  </li></div></section><footer class=\\"footer\\"><span class=\\"todo-count\\"><strong>2</strong> todo</span></footer></section>"
`;
